spec/features/groceries_spec.rb

```diff
@@ -9,16 +9,21 @@
 
       let(:new_item_name) { 'blueberries' }
 
-      it 'allows adding an item' do
+      it 'allows adding an item, deleting an item, undoing the deletion, and checking in a shopping trip' do
         visit groceries_path
 
         store = user.stores.reorder(viewed_at: :desc).first!
         expect(page).to have_text(store.name)
         expect(page).to have_button('Check in items')
 
         needed_item = store.items.needed.first!
+        unneeded_item = store.items.unneeded.first!
         expect(page).to have_text(/#{needed_item.name} +\(#{needed_item.needed}\)/)
 
+        # NOTE: When running specs with the development vite server, the
+        # following line triggers two warnings, which I'm pretty sure are caused
+        # by a bug in cuprite (namely that focus and blur events are of type
+        # `Event` rather than `FocusEvent`).
```

Comment:
The bug is fixed by changing this line of code: https://github.com/rubycdp/cuprite/blob/v0.15.1/lib/capybara/cuprite/javascripts/index.js#L419

from this

```js
event = this.obtainEvent(name);
```

to this

```js
event = document.createEvent("FocusEvent");
event.initEvent(name, true, true)
```